src/gui/qgsmaskingwidget.cpp

```diff
@@ -28,29 +28,54 @@
 #include "qgsvectorlayerutils.h"
 #include "qgsmasksymbollayer.h"
 #include "qgsvectorlayerlabeling.h"
+#include "qgsmessagebaritem.h"
 
 QgsMaskingWidget::QgsMaskingWidget( QWidget *parent ) :
   QgsPanelWidget( parent )
 {
   setupUi( this );
+}
+
+void QgsMaskingWidget::onSelectionChanged()
+{
+  // display message if configuration is not consistent
+  bool printMessage =
+    ( mMaskTargetsWidget->selection().empty() && !mMaskSourcesWidget->selection().empty() )
+    || ( !mMaskTargetsWidget->selection().empty() && mMaskSourcesWidget->selection().empty() );
 
-  connect( mMaskTargetsWidget, &QgsSymbolLayerSelectionWidget::changed, this, [&]()
+  if ( mMessageBarItem && !printMessage )
   {
-    mMaskSourcesWidget->setEnabled( ! mMaskTargetsWidget->selection().isEmpty() );
-    emit widgetChanged();
-  } );
-  connect( mMaskSourcesWidget, &QgsMaskSourceSelectionWidget::changed, this, [&]()
+    mMessageBar->popWidget( mMessageBarItem );
+    delete mMessageBarItem;
```

Comment:
What happens if the user clicks the `X` on the item? is there a risk of a crash?

Reply:
yes. I added a QPointer